### ISSUE TITLE
Enabling termguicolor breaks xterm colors

### ISSUE THREAD

**Issue by dannycolin** [joshdick/onedark.vim#188]
- [x] I have carefully read and followed the [Installation section of the README](https://github.com/joshdick/onedark.vim#installation)
- [x] I have searched through the [closed issues](https://github.com/joshdick/onedark.vim/issues?q=is%3Aissue+is%3Aclosed) to see if there is an existing solution for my issue

### Terminal Emulator Info

Terminal emulator(s) and their version(s) used, if applicable: `XTerm(334)` on Fedora 30.

### Output From `nvim --version`

```
NVIM v0.3.8
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -Wconversion -Wp,-U_FORTIFY_SOURCE -Wp,-D_FORTIFY_SOURCE=1 -O2 -g -DMIN_LOG_LEVEL=3 -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -Wno-array-bounds -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/builddir/build/BUILD/neovim-0.3.8/build/config -I/builddir/build/BUILD/neovim-0.3.8/src -I/usr/include -I/builddir/build/BUILD/neovim-0.3.8/build/src/nvim/auto -I/builddir/build/BUILD/neovim-0.3.8/build/include
Compiled by mockbuild

Features: +acl +iconv -jemalloc +tui 
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

### Issue Description

If `set termguicolors` is set and I exit nvim, my xterm colors are messed up. The background color where there's no text is white (color15) instead of black (color0). (See: https://imgur.com/fyGdmH2)

Note: Running the True Color script works. So my terminal definitely supports it.

**Comment by joshdick** [comment]
This is an interesting one for sure -- this is the first report I've received about a problem after **exiting** vim! 😆 

I don't have much to go on here. Is this only happening with onedark.vim, or with any colorscheme as long as `termguicolors` is set? If the latter, you could try upgrading to the latest version of Neovim; I see from your report that you're running 0.3.8 but the latest version is 0.4.2.

**Comment by dannycolin** [comment]
> Is this only happening with onedark.vim, or with any colorscheme as long as termguicolors is set?

I've tested with other colorscheme and fount out that it only happens with non-default colorscheme. 

> I see from your report that you're running 0.3.8 but the latest version is 0.4.2.

I have the same problem with 0.4.2, atest master commit on neovim and Vim 8.1.2102.

Seeing that it happens with both neovim and vim, I've investigate a bit more to see if the problem wasn't Xterm. First test I did has to try a different terminal like *alacritty* and I couldn't reproduce the error. So I thought I have on the right track thinking it was xterm. However, the exact same *xterm* version (334) works perfectly on Debian Stable but not on my Fedora 30 laptop. So, I don't know what to think about it :smile:.
